src/crawlee/_autoscaling/autoscaled_pool.py

```diff
@@ -32,6 +33,7 @@ def __init__(self) -> None:
         self.result: asyncio.Future = asyncio.Future()
 
 
+@docs_group('Classes')
```

Comment:
I'd make a string enum for the possible groups... So that we get hints from the IDE.

Reply:
It would also let us specify the order of groups that way